pkg/controllerutils/patch_test.go

```diff
@@ -185,6 +208,10 @@ var _ = Describe("Patch", func() {
 
 				gomock.InOrder(
 					c.EXPECT().Create(ctx, obj).Return(apierrors.NewAlreadyExists(schema.GroupResource{}, "")),
+					c.EXPECT().Get(ctx, client.ObjectKeyFromObject(obj), obj).DoAndReturn(func(_ context.Context, _ client.ObjectKey, objToReturn *corev1.ServiceAccount) error {
+						obj.DeepCopyInto(objToReturn)
+						return nil
```

Comment:
Can you add the following to assert that the reset logic is working properly?

```suggestion
						obj.DeepCopyInto(objToReturn)
												// mimic behavior of json decoder: if optional fields are not set on the server, it won't unset them on the in-memory object.
						// In this case: the mutate func was already called for the create call, so the in-memory object will have the labels, but the
						// get call won't remove them.
						objToReturn.Labels = objCopy.Labels
						return nil
```

With this, the test will fail, if the reset func is not called. Currently, all tests will go green, when you remove the reset call.

Reply:
I wanted to try out the "allow edits by maintainers" feature and pushed the change as another commit myself, hope you don't mind 😉 
Let me know what you think :)

Reply:
Why are the tests failing now after your change?

Reply:
Well, I figured, my change didn't make any sense 🙈 
I pushed another commit that asserts that our reset logic is working, so what I originally wanted to do :)

Reply:
OK, this looks good. Who approves this PR now? 😉

Reply:
I guess, we both? 🤔😁

Reply:
![image](https://user-images.githubusercontent.com/19169361/121013956-bbd2a500-c799-11eb-82e8-851728a75568.png)

approve :)